scheduler/scheduler-server/src/main/java/org/ow2/proactive/scheduler/synchronization/AOSynchronization.java

```diff
@@ -0,0 +1,824 @@
+            for (Map.Entry<String, T> entry : chosenChannel.entrySet()) {
+                currentKey = entry.getKey();
+                T value = chosenChannel.get(currentKey);
+                if (evaluateClosure(predicate, BiPredicate.class).test(currentKey, value)) {
+
+                    T answer = chosenChannel.compute(currentKey,
+                                                     (BiFunction<? super String, ? super T, ? extends T>) evaluateClosure(thenRemappingFunction,
+                                                                                                                          BiFunction.class));
+                    logWithContext(taskid,
+                                   channel,
+                                   "Run conditionalReplaceAll(Then) on key '" + currentKey + "' result is: " + answer,
+                                   Level.DEBUG);
+                } else {
+                    T answer = chosenChannel.compute(currentKey,
+                                                     (BiFunction<? super String, ? super T, ? extends T>) evaluateClosure(elseRemappingFunction,
+                                                                                                                          BiFunction.class));
+                    logWithContext(taskid,
+                                   channel,
+                                   "Run conditionalReplaceAll(Else) on key '" + currentKey + "' result is: " + answer,
+                                   Level.DEBUG);
+
+                }
+            }
+            commitIfNeeded(channel);
+
+        } catch (InvalidChannelException | CompilationException | IOException e) {
+            throw e;
+        } catch (Throwable e) {
+            throw new ClosureEvaluationException("Exception when evaluating closure : " +
+                                                 StackTraceUtil.getStackTrace(e));
+        }
+    }
+
+    @Override
+    public <T extends Serializable> boolean waitUntil(String originator, TaskId taskid, String channel, String key,
+            String predicate) throws InvalidChannelException, CompilationException {
+        Channel<T> chosenChannel = getChannel(channel);
+        T value = chosenChannel.get(key);
+        try {
+            return evaluateClosure(predicate, BiPredicate.class).test(key, value);
+        } catch (CompilationException e) {
+            throw e;
+        } catch (Throwable e) {
+            throw new ClosureEvaluationException("Exception when evaluating closure : " +
+                                                 StackTraceUtil.getStackTrace(e));
+        }
+    }
+
+    @Override
+    public <T extends Serializable> PredicateActionResult<T> waitUntilThen(String originator, TaskId taskid,
+            String channel, String key, String predicate, String thenRemappingFunction)
+            throws InvalidChannelException, CompilationException, IOException {
```

Comment:
![MINOR](https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR') Remove the redundant '!unknownSymbol!' thrown exception declaration(s). [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3ARedundantThrowsDeclarationCheck)